File: Strings/__2000__/2108/2108.php
<?php 

    $maior = 0;
    $maior_palavra = "";
    $line = readline();

    while($line != "0"){

        $resultado = "";
        $words = explode(' ', $line);

        for($i = 0; $i < count($words); $i++){
            if($resultado != ""){ $resultado .= "-"; }

            if (strlen($words[$i]) >= $maior){
                $maior = strlen($words[$i]);
                $maior_palavra = $words[$i];
            }

            $resultado .= strlen($words[$i]);
        }

        echo "$resultado\n";
        $line = readline();
    }
    echo "\n";
    echo "The biggest word: $maior_palavra\n";
?>

File: Strings/__2000__/2108/2108.java
import java.util.Scanner;
import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException {
        Scanner in = new Scanner(System.in);

        int maior = 0;
        String maior_palavra = "";
        String line = in.nextLine();

        while(!line.equals("0")){

            String resultado = "";
            String[] words = line.split(' ');

            for(int i = 0; i < words.length; i++){
                if(!resultado.isEmpty()){ resultado += "-"; }

                if (words[i].length() >= maior){
                    maior = words[i].length();
                    maior_palavra = words[i];
                }

                resultado += words[i].length();
            }

            System.out.println(resultado);
            line = in.nextLine();
        }
        System.out.println("");
        System.out.println("The biggest word: " + maior_palavra);
        in.close();
    }
}